starter_code/elevator.js

```diff
@@ -1,19 +1,46 @@
+const Person = require('./person.js');
+
 class Elevator {
-  constructor(){
-    this.floor      = 0;
-    this.MAXFLOOR   = 10;
-    this.requests   = [];
+  constructor () {
+    this.floor = 0;
+    this.MAXFLOOR = 10;
+    this.requests = [];
+    this.direction = '';
   }
 
-  start() { }
-  stop() { }
-  update() { }
-  _passengersEnter() { }
-  _passengersLeave() { }
-  floorUp() { }
-  floorDown() { }
-  call() { }
-  log() { }
+  start () {
+    this.myStart = setInterval(() => this.update(), 1000);
+  }
+  stop () {
+    clearInterval(this.myStart);
+  }
+  update () {
+    if (this.direction === 'up') {
+      this.floorUp();
+    } else {
+      this.floorDown();
+    }
+    this.log();
+  }
+  _passengersEnter () { }
+  _passengersLeave () { }
+  floorUp () {
+    if (this.floor < this.MAXFLOOR) {
+      this.floor++;
+    }
+  }
+  floorDown () {
+    if (this.floor > 0) {
+      this.floor--;
+    }
+  }
+  call () {
+    this.passenger = new Person(this.name, this.originFloor, this.destinationFloor);
+    this.requests.push(this.passenger);
+  }
```

Comment:
Hmm nice approach. Better create passengers in your index.js and pass them as an argument of call(). Something like
```
//index
let byron = new Person("Byron", 7, 1);
let elevator = new Elevator();
elevator.call(byron);
```
```
//elevator
call(person) { bla bla bla}
```